onnxruntime/test/providers/cpu/tensor/concat_op_test.cc

```diff
@@ -111,6 +111,24 @@ TEST(ConcatOpTest, Concat2D_3) {
             kNnapiExecutionProvider});   // NNAPI: concat does not support 0 size input
 }
 
+TEST(ConcatOpTest, Concat2D_4) {
+  OpTester test("Concat");
+  test.AddAttribute("axis", int64_t{1});
+
+  std::vector<int64_t> dims{4, 1};
+  std::vector<std::string> dim_params{"batch", "seq"};
+  test.AddInput<float>("input1", dims, {11.0f, 21.0f, 31.0f, 41.0f});
```

Comment:
nit: would be helpful to put a comment explaining why we do something so obtusely different for one input. that way the next developer to look at the code knows it's intentional and why it's being done, as it's unlikely they're aware of the specific processing in the NNAPI EP being tested.